StereoKitC/asset_types/model.cpp

```diff
@@ -273,25 +273,31 @@ bounds_t model_get_bounds(model_t model) {
 ///////////////////////////////////////////
 
 bool32_t model_ray_intersect(model_t model, ray_t model_space_ray, cull_ cull_mode, ray_t *out_pt) {
+	model_node_id intersected_model_nodeid = (model_node_id)-1;
+	return model_node_ray_intersect(model, model_space_ray, cull_mode, out_pt, &intersected_model_nodeid);
+}
+
+bool32_t model_node_ray_intersect(model_t model, ray_t model_space_ray, cull_ cull_mode, ray_t* out_pt, model_node_id* intersected_model_nodeid) {
 	vec3 bounds_at;
 	if (!bounds_ray_intersect(model->bounds, model_space_ray, &bounds_at))
 		return false;
 
 	float closest = FLT_MAX;
 	*out_pt = {};
```

Comment:
`out_opt_node_id` would also need to be provided a default value of `-1` here, if not `nullptr`